src/api.js

```diff
@@ -20,7 +20,7 @@ const SessionExpiresInHeader = 'X-Session-Expires-In';
 
 let sessionExpiresAt = createGlobalstate({expiry: null});
 
-const updateSesionExpiry = seconds => {
+export const updateSesionExpiry = seconds => {
```

Comment:
At first I wasn't delighted to see this exported, but I guess that state pool was exposed anyway.

My concern was that I say myself chasing timeouts because somewhere, some component mutates the expiry time outside of the fetch requests flow.

Maybe fix the typo... `updateSessionExpiry`

Reply:
Yeah it hurt me too, I wish there was a mechanism to limit exports to tests :(

Re-implementing it directly on the state pool seemed like a bigger evil.

> My concern was that I say myself chasing timeouts because somewhere, some component mutates the expiry time outside of the fetch requests flow.

This already is a "problem" - the session cookie already is a global state so managing the expiry in the app also has to be global. It's a side effect of every API call. I do have a draft issue on the board to refactor the session expiry component into it's own explicit route to simplify/decouple things a bit more.

Will fix the typo!